File: app/src/main/java/com/digzdigital/divinitytoday/ui/devlist/DevListContract.java
package com.digzdigital.divinitytoday.ui.devlist;

import com.digzdigital.divinitytoday.data.Devotional;

import java.util.ArrayList;


interface DevListContract {
    interface View {

        void doRest(final ArrayList<Devotional> devotionals);

        void showProgressDialog();

        void dismissProgressDialog();

        void makeToast(String message);

    }

    interface Presenter {

        void setView(DevListActivity view);

        void loadDevotionals(int endpoint);

        int getDevSize();

    }
}

File: app/src/main/java/com/digzdigital/divinitytoday/ui/devlist/DevListContract.kt
package com.digzdigital.divinitytoday.ui.devlist

import com.digzdigital.divinitytoday.data.model.Devotional
import com.digzdigital.divinitytoday.ui.base.BasePresenter
import com.digzdigital.divinitytoday.ui.base.BaseView

interface DevListContract {
    interface View : BaseView {

        fun showDevotionalsAndAds(devotionals: List<Devotional>)

        fun showMoreDevotionals(devotionals: List<Devotional>)

        fun showProgressDialog()

        fun dismissProgressDialog()

        fun makeToast(message: String)

        fun setRefreshingOff()
    }

    interface Presenter : BasePresenter<View> {

        fun loadDevotionals()

        fun loadMoreDevotionals(endpoint: Int = 0)

        fun refreshDevotionals()

        fun addDevotionalToBookmarks(devotional: Devotional)

        fun removeDevotionalFromBookmarks(devotional: Devotional)
    }
}

